packages/core/components/NestedDropdown/NestedDropdown.tsx

```diff
@@ -0,0 +1,251 @@
+          itemToFocusOnAfterKeyUp.focus()
+        } else if (className === 'nested-dropdown__group--hidden') {
+          // If Tier One is collapsed, move to next Tier One or move focus back to the top Input
+          const itemToFocusOnAfterKeyUp = nextSibling ?? searchInput.current
+          itemToFocusOnAfterKeyUp.focus()
+        } else {
+          // If Tier One is open, move focus to Tier Two
+          lastChild?.firstChild?.focus()
+        }
+        break
+      }
+
+      case 'ArrowUp': {
+        if (nodeName === 'INPUT') {
+          setIsListOpened(true)
+          // Move focus from Input to collapsed Tier Two or last item of Tier Two
+          if (searchDropdown.current.lastChild.className === 'nested-dropdown__group--hidden') {
+            searchDropdown.current.lastChild.focus()
+          } else {
+            searchDropdown.current.lastChild.lastChild.lastChild.focus()
+          }
+        } else if (className === 'selectableItem') {
+          // Move focus to previous Tier Two or Move focus to current Tier One
+          const itemToFocusOnAfterKeyUp = previousSibling ?? parentNode.parentNode
+          itemToFocusOnAfterKeyUp.focus()
+        } else if (previousSibling) {
+          // Move focus to previous collapsed Tier One or Move focus from Tier One to the last of the previous Tier Two's items
+          const itemToFocusOnAfterKeyUp = previousSibling.className === 'nested-dropdown__group--hidden' ? previousSibling : previousSibling.lastChild.lastChild
+          itemToFocusOnAfterKeyUp.focus()
+          // } else if (e.target !== searchInput.current) {
+        } else {
+          // Move focus from top of the dropdown to Input
+          searchInput?.current.focus()
+        }
+        break
+      }
+
+      case 'ArrowLeft': {
+        if (nodeName === 'INPUT') {
+          setIsListOpened(false)
+        }
+        break
+      }
+
+      case 'ArrowRight': {
+        if (nodeName === 'INPUT') {
+          setIsListOpened(true)
+        }
+        break
+      }
+
+      case 'Escape':
+        {
+          setIsListOpened(false)
+          searchInput.current.focus()
+        }
+        break
+    }
+  }
+
+  const filterOptions: Record<string, (string | number)[]> = useMemo(() => {
+    if (!userSearchTerm) return optsMemo
+    const newOptions: Record<string, (string | number)[]> = {}
+    const newRegex = new RegExp(`^${userSearchTerm}`, 'i')
+    for (const tierOne in optsMemo) {
+      const newSecondTierOptions = optsMemo[tierOne].filter(tierTwo => String(tierTwo).match(newRegex))
+      if (newSecondTierOptions.length > 0) {
+        newOptions[tierOne] = newSecondTierOptions
+      } else if (tierOne.match(newRegex)) {
+        newOptions[tierOne] = [...optsMemo[tierOne]]
+      }
+    }
+    return newOptions
+  }, [userSearchTerm])
+
+  const filterOptionsKeys = Object.keys(filterOptions)
+
+  const handleSearchTermChange = (e: { target: { value: any } }) => {
```

Comment:
why de-structure the params when you aren't using them below?

Reply:
de-structure removed